server/graphql_schemas/event/schema.py

```diff
@@ -23,43 +25,81 @@ class Input:
         date = graphene.String(required=True)
         time = graphene.String(required=False)
         featured_image = graphene.String(required=False)
-        social_event_id = graphene.Int(required=False)
+        social_event_id = graphene.String(required=False)
 
     new_event = graphene.Field(EventNode)
 
     @classmethod
     def mutate_and_get_payload(cls, root, info, **input):
-        user = info.context.user
         social_event_id = input.get('social_event_id')
-        social_event = Category.objects.get(id=int(social_event_id))
-        andela_user_profile = AndelaUserProfile.objects.get(user_id=user.id)
-        new_event = Event(
-          title=input.get('title'),
-          description=input.get('description'),
-          venue=input.get('venue'),
-          date=input.get('date'),
-          time=input.get('time'),
-          featured_image=input.get('featured_image'),
-          creator=andela_user_profile,
-          social_event=social_event
-        )
-        new_event.save()
+        try:
+            social_event = Category.objects.get(
+                id=from_global_id(social_event_id)[1])
+            user_profile = AndelaUserProfile.objects.get(
+                user=info.context.user
+                )
+            new_event = Event(
+                title=input.get('title'),
+                description=input.get('description'),
+                venue=input.get('venue'),
+                date=input.get('date'),
+                time=input.get('time'),
+                featured_image=input.get('featured_image'),
+                creator=user_profile,
+                social_event=social_event
+            )
+            new_event.save()
+
+        except ValueError as e:
+            raise GraphQLError("An Error occured. \n{}".format(e))
 
         return cls(new_event=new_event)
 
 
+class UpdateEvent(relay.ClientIDMutation):
+
+    class Input:
+        title = graphene.String()
+        description = graphene.String()
+        venue = graphene.String()
+        date = graphene.String()
+        time = graphene.String()
+        featured_image = graphene.String()
+        social_event_id = graphene.String(required=True)
+
+    updated_event = graphene.Field(EventNode)
+    action_message = graphene.String()
+
+    @classmethod
+    def mutate_and_get_payload(cls, root, info, **input):
+
+        try:
+            user = AndelaUserProfile.objects.get(user=info.context.user)
+            event_instance = Event.objects.get(
+                id=from_global_id(input.get('social_event_id'))[1]
+            )
+            if event_instance.creator != user \
+                    and not info.context.user.is_superuser:
+                raise GraphQLError(
+                    "You are not authorized to edit this event.")
+
+            if event_instance:
+                updated_event = update_instance(
```

Comment:
@parseendavid This line of code is now failing because of social_event_id is a string and we use from_global_id to get the actual Id value. so there's a need to tweak `update_instance method` or we change the social_event_id value in the input before passing it to `update_instance` helper function.